### PR TITLE
[Helm] Pick up the symbol at point when do smart search with "rg"

### PR DESCRIPTION
**Pull request by sunlin7**
The `spacemacs/helm-project-smart-do-search' will pick up the symbol at point
when searching with "grep" backend, but pick up nothing when searching with "rg"
backend, here we make the "rg" backend has similar behavior as default backend.

### CODE REVIEW

**Comment by sunlin7** [comment]
@smile13241324 Could you please review this change? Thank you.

**Comment by smile13241324** [comment]
I am not sure whether prompting the symbol at point to your smart search is not a controversial change. Let me discuss this with the other maintainers first.

**Comment by sunlin7** [comment]
Okay, look forward your good news. The change is try to make the  "rg" backend behavior to follow the default backend (grep/ack).

**Comment by nixmaniack** [comment]
I'm not sure why this change is required.

If you check [here](https://github.com/syl20bnr/spacemacs/blob/2049817b1281dbda1286771a99498aa6ea4696bc/doc/DOCUMENTATION.org#searching-in-a-project), it'll give you a bit more idea.

In brief, `SPC /` or `SPC s p` which is bound to  `spacemacs/helm-project-smart-do-search` allows you to `search project` with an input prompt.
And, `SPC *` or `SPC s P` which is bound to `spacemacs/helm-project-smart-do-search-region-or-symbol` allows you to search project `with symbol/word under point`.

This all pieces together with auto tool selection mechanism based on the availability of the `search tool` (rg, ag etc. defined with `dotspacemacs-search-tools`) on the system. So, changing the function will essentially break `SPC /` and functions dependent on it. 

Another disclaimer, I use `ivy` instead of `helm` but the underlying mechanism remains same as far as I understand. So, unless `SPC *` related functionality is broken in `helm`, this change is not required. 

TL;DR: What the OP wants is already available with `SPC *` or `SPC s P` or if you need specific tool based search e.g. `rg` use `SPC r P`.

**Comment by sunlin7** [comment]
@nixmaniack Thank you for your explanation. This change aim to make the "rg" backend follow the  beharior of default "grep" backend for completion with heml.
With the "helm" layer enabled, the function `spacemacs/helm-project-smart-do-search'  will pick up the symbol at point with default backend while it pick up nothing when there is "rg" in the path, it's confused me quite a while for same spacemacs configurationes on two machines, one has "rg" but another didn't.
And the change acturly for helm, it made the function `spacemacs/helm-files-do-rg' act like `spacemacs/helm-files-do-grep', didn't affect ivy.

**Comment by nixmaniack** [comment]
As explained in the previous comment,
`SPC /` **shouldn't** fill in the symbol at point
`SPC *` **should** fill in the symbol at point 
the search tool doesn't matter, the functionality/convention should be adhered to.

If having `grep` as search tool causes the helm prompt to fill in the symbol at point for `SPC /`, then that's a bug with "helm smart search with grep" and we should be fix that. 

Could you tell us which keybinding you're invoking and what is the expected behaviour?

**Comment by sunlin7** [comment]
@nixmaniack Okay. Close this PR as innocent. Thank you.
It's `M-m /`, `spacemacs/helm-project-smart-do-search`.

**Comment by nixmaniack** [comment]
You may want to create an issue for the incorrect behaviour of helm grep.

**Comment by smile13241324** [comment]
I am not sure why helm-grep adds symbol under point to the search input. I have double checked the code and it clearly gives "nil" to helm-grep-1. Maybe there is an issue with helm-grep? I will need to investigate a bit further for this.
